Sigfox-sensit-v2.js

```diff
@@ -0,0 +1,286 @@
+  return [temperatureMsb, batteryLsb];
+}
+//-------------------------------------------------------
+
+// Byte 3 - variables extraction ------------------------
+
+// This byte depends on the mode
+function byte3VE(byte3, mode) {
+  return BYTE3_FUNCTIONS[mode](byte3);
+}
+
+// Byte 3: Case for Temperature & Humidity mode
+function byte3Case1(byte3) {
+  const temperatureLsb = byte3.slice(2, 8);
+  return { temperatureLsb };
+}
+
+// Byte 3: Case for Light mode
+function byte3Case2(byte3) {
+  const lightMask = byte3.slice(0, 2);
+  const lightValue = byte3.slice(2, 8);
+  return { lightMask, lightValue };
+}
+//-------------------------------------------------------
+
+// Byte 4 - variables extraction ------------------------
+
+// This byte depends on the mode
+function byte4VE(byte4, mode) {
+  return BYTE4_FUNCTIONS[mode](byte4);
+}
+
+// Byte 4: Case for Temperature & Humidity mode
+function byte4Case1(byte4) {
+  const humidity = byte4;
+  return { humidity };
+}
+
+// Byte 4: Case for Light mode
+function byte4Case2(byte4) {
+  const alertCounter = byte4; // Number of events that happened
+  return { alertCounter };
+}
+//---------------------------------------------------------------------
+
+//---------------------------------------------------------------------
+// Functions to get variables' final values
+
+// Battery voltage --------------------------------------
+function batteryVoltage(batteryMsb, batteryLsb) {
+  let batteryLevel = `${batteryMsb}${batteryLsb}`;
+  batteryLevel = parseInt(batteryLevel, 2);
+  batteryLevel = (batteryLevel * 0.05) + 2.7; // Volts
+  return (batteryLevel);
+}
+//-------------------------------------------------------
+
+// Temperature & Humidity mode - data decoding ----------
+function modeTemperature(varsToExtract) {
+  // Expected input: [temperatureMsb, temperatureLsb, humidity]
+  const temperatureMsb = varsToExtract[0];
+  const temperatureLsb = varsToExtract[1];
+  const humidity = varsToExtract[2];
+  let fTemperature; // Final temperature value
+  let fHumidity; // Final humidity value
+
+  fTemperature = `${temperatureMsb}${temperatureLsb}`; // 10 bits
+  fTemperature = parseInt(fTemperature, 2);
+  fTemperature = ((fTemperature - 200) / 8); // °C
+
+  fHumidity = parseInt(humidity, 2);
+  fHumidity /= 2; // %
+
+  const decodedOutput = {
+    temperature: fTemperature,
+    humidity: fHumidity,
+  };
+  return (decodedOutput);
+}
+//-------------------------------------------------------
+
+// Light mode - data decoding ---------------------------
+function modeLight(varsToExtract) {
+  // Expected input: ['lightMask', 'lightValue', 'alertCounter']
+  const lightMask = varsToExtract[0];
+  const lightValue = varsToExtract[1];
+  const fLightValue = LIGHT_VALUE[lightMask](lightValue);
+  const decodedOutput = { light: fLightValue };
+  return decodedOutput;
+}
+
+function lightCase0(lightValue) {
+  const light = parseInt(lightValue, 2);
+  const fLightValue = light / 96; // lux
```

Comment:
store the value 96 in a variable that describes why are you using this constant, i.e:

const adc_conversion_factor = 96;  // Taken from this doc (link)
 const fLightValue = light / adc_conversion_factor; // lux